avocado/core/job.py

```diff
@@ -547,7 +496,7 @@ def run_tests(self):
         jobdata.record(self.config, self.logdir, variant, sys.argv)
         summary = self.test_runner.run_suite(self,
                                              self.result,
-                                             self.test_suite,
+                                             self.test_suite.tests,
```

Comment:
I understand this is the introduction of the `TestSuite` class and is the way it is for compatibility. As an evolution, I see the `run_suite` still receiving as an argument the `self.test_suite` and handling whatever is needed inside it.

Reply:
Agree.